### PR TITLE
[sig-monitoring] Create token for prometheus-k8s service account

### PR DESCRIPTION
**Pull request by brianmcarey**
**What this PR does / why we need it**:
Later versions of kubernetes do not automatically generate tokens for service accounts[1]. This leads to the monitoring tests timing out due to the authorization token not being in place [2].

This change creates a service account token which allows the tests to complete successfully on versions later than 1.23. Tested locally against the 1.26 kubevirtci provider.

[1] https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-an-api-token-for-a-serviceaccount 
[2] https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/2553/rehearsal-pull-kubevirt-e2e-k8s-1.26-sig-monitoring/1615267361609224192

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by enp0s3 — COMMENTED** [review]
/approve

File: tests/monitoring/prometheus_utils.go [hunk 1]
```diff
@@ -205,19 +205,30 @@ func doHttpRequest(url string, endpoint string, token string) *http.Response {
...
 		sa, err := cli.CoreV1().ServiceAccounts(monitoringNs).Get(context.TODO(), "prometheus-k8s", metav1.GetOptions{})
 		if err != nil {
 			return false
 		}
-		for _, secret := range sa.Secrets {
-			if strings.HasPrefix(secret.Name, "prometheus-k8s-token") {
-				secretName = secret.Name
-			}
-		}
+		secretName := fmt.Sprintf("%s-%s-token", sa.Name, monitoringNs)
```
Comment (xpivarc) [inline tests/monitoring/prometheus_utils.go:212]: Isn't the previous API call redundant?

Comment (brianmcarey) [reply tests/monitoring/prometheus_utils.go:212]: Yes I think I can remove and just use the string.

Comment (enp0s3) [reply tests/monitoring/prometheus_utils.go]: Is this token are going to be deleted upon test suite teardown?

**Comment by xpivarc** [comment]
/lgtm
Nice work!

**Pull request merged**